lib/appdmg.js

```diff
@@ -77,13 +77,19 @@ module.exports = exports = function (options) {
    **/
 
   pipeline.addStep('Looking for target', function (next) {
-    fs.exists(global.target, function (exists) {
-      if (exists) {
-        next(new Error('Target already exists'))
+    pipeline.addCleanupStep('unlink-target', 'Removing target image', function (next, hasErrored) {
+      if (hasErrored) {
+        fs.unlink(global.target, next)
       } else {
         next(null)
       }
     })
+
+    fs.open(global.target, 'wx', function (error) {
+      if (error) return next(new Error('Target already exists'))
```

Comment:
If you are going to do nice error reporting you need to look at `error.code`, there are a number of other reasons for this to error